steve/util.py

```diff
@@ -378,10 +378,14 @@ def out(*output, **kw):
     sys.stdout.write(output + '\n')
 
 
-def load_json_files(config):
+def load_json_files(config, path=None):
```

Comment:
I think you're overcomplexifying this. I don't think there's a need to support multiple data directories for a project. I don't think it makes sense to add the additional complexity to support that.

Generally, I find what you did overly complex and I don't think I understand what the requirements you're working around are. Maybe adding tests covering the cases you're trying to handle would help?